### ISSUE TITLE
Legacy firewall rules endpoint returns api.err.InvalidObject for invited/non-owner admin account despite Full Management role

### ISSUE THREAD

**Issue by mbauwens** [sirkirby/unifi-network-rules#158]
## Debug Logging

- [x] debug logging is enabled in the configuration.

## Describe the bug

For the site owner's own account, `get_legacy_firewall_rules()` works correctly and returns all
legacy LAN firewall rules that exist on the site. For a second, separate account that was
invited/added to the same site (not the original owner) and granted the Network application's
highest "Full Management" role, the identical request (`GET /firewallrule` against the local
UniFi OS controller) instead returns `meta.rc: "error"`, `meta.msg: "api.err.InvalidObject"`.

The integration currently treats this response as "legacy firewall rules not available - device
likely using zone-based firewalls" and silently returns an empty list. In this case that's a
false negative: the rules genuinely exist and are readable by the owner account on the exact same
controller/site/firmware.

I confirmed this by calling the same endpoint directly (outside of Home Assistant) with each
account's own credentials: the owner account gets a normal 200 with the full rule list; the
invited account gets `api.err.InvalidObject` on this one endpoint specifically, while every other
resource type I checked (WLANs, traffic routes, VPN clients/servers, networks, devices) loads
correctly for that same invited account.

I also tried granting that invited account the highest console-level "Full Admin" permission
available in the UniFi app's permission picker, in addition to the Network app's "Full
Management" role — no change in behavior.

## To Reproduce

1. On a UniFi OS console that has at least one legacy (non-zone-based) LAN firewall rule
   configured, create a second local/invited admin account (not the original site owner) and
   grant it the highest available role for the Network application.
2. Set up this integration in Home Assistant using that second account's credentials.
3. Enable debug logging and trigger a refresh (`unifi_network_rules.refresh`).
4. Check the debug log for the legacy firewall rules fetch.

## Expected behavior

`get_legacy_firewall_rules()` should either return the actual rule list, or surface a
distinguishable error/warning, rather than silently collapsing a permission-shaped
`api.err.InvalidObject` response into the same "not available / likely zone-based" bucket as a
genuine unsupported-hardware case. These are very different situations that currently look
identical from the integration's own logs.

## Home Assistant Logs

```
DEBUG ... Creating API request: method=GET, endpoint=/firewallrule, is_v2=False
DEBUG ... Legacy firewall rules not available - device likely using zone-based firewalls
INFO  ... Fetched 0 legacy_firewall_rules
```

## Additional context

- `firewall_policies` and `firewall_zone` are also (correctly) empty for this account/site, since
  the site is on the legacy firewall model, not zone-based — confirmed the controller's own UI
  still shows and uses the legacy Firewall Rules page. So this is specifically about the
  owner-vs-invited-account distinction, not the legacy-vs-zone-based distinction the current debug
  message implies.
- As a workaround for my own use case, I ended up bypassing this integration for just the
  affected rules: a UniFi "Network API" key (Settings > Control Plane > Integrations) tied to the
  owner account, called directly via Home Assistant's built-in RESTful `switch` platform. That
  confirms the controller and endpoint themselves work fine — the gap is specifically in how this
  integration's invited account interacts with the controller's permission model for this one
  legacy object type.
- Might be worth distinguishing an `api.err.InvalidObject` response (looks like a real
  capability/permission signal) from a genuinely unsupported-hardware case, and/or logging a
  warning rather than debug when a legacy-rules fetch errors out, so this doesn't silently look
  like "0 rules because there are none."

---
*Note: this issue was drafted by Claude Code, checked and approved by me before posting.*
